qemu/tests/guest_iommu_test.py

```diff
@@ -0,0 +1,128 @@
+from virttest import cpu
+from virttest import utils_test
+from virttest import utils_disk
+from provider.storage_benchmark import generate_instance
+
+
+def check_data_disks(test, params, env, vm, session):
+    """
+    Check guest data disks (except image1)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    :param vm: VM object
+    :param session: VM session
+    """
+    def _get_mount_points():
+        """ Get data disk mount point(s) """
+        mount_points = []
+        os_type = params["os_type"]
+        if os_type == "linux":
+            mounts = session.cmd_output_safe('cat /proc/mounts | grep /dev/')
+            for img in image_list:
+                size = params["image_size_%s" % img]
+                img_param = params["blk_extra_params_%s" % img].split('=')[1]
+                drive_path = utils_misc.get_linux_drive_path(session, img_param)
+                if not drive_path:
+                    test.error("Failed to get drive path of '%s'" % img)
+                did = drive_path[5:]
+                for mp in re.finditer(r'/dev/%s\d+\s+(\S+)\s+' % did, mounts):
+                    mount_points.append(mp.group(1))
+                else:
+                    mp = utils_disk.configure_empty_linux_disk(session,
+                                                               did, size)
+                    mount_points.extend(mp)
+        elif os_type == "windows":
+            size_record = []
+            for img in image_list:
+                size = params["image_size_%s" % img]
+                if size in size_record:
+                    continue
+                size_record.append(size)
+                disks = utils_disk.get_windows_disks_index(session, size)
+                if not disks:
+                    test.fail("Fail to list image %s" % img)
+                if not utils_disk.update_windows_disk_attributes(session,
+                                                                 disks):
+                    test.fail("Failed to update windows disk attributes")
+                for disk in disks:
+                    d_letter = utils_disk.configure_empty_windows_disk(session,
+                                                                       disk,
+                                                                       size)
+                if not d_letter:
+                    test.fail("Fail to format disks")
+                mount_points.extend(d_letter)
+        else:
+            test.cancel("Unsupported OS type '%s'" % os_type)
+        return mount_points
+
+    image_list = params.objects("images")[1:]
+    image_num = len(image_list)
+
+    error_context.context("Check data disks in monitor!", logging.info)
+    monitor_info_block = vm.monitor.info_block(False)
+    blocks = ','.join(monitor_info_block.keys())
+    for image in image_list:
+        if image not in blocks:
```

Comment:
This approach is not always right, suppose that you have two images: `stg1` and `stg11`, the result would both be `False` while `blocks` is `"disk_stg11"`.

Reply:
Updated, thanks!